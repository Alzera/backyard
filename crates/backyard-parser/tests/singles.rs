use backyard_parser::{ parse, parse_eval };

#[test]
fn test_break() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "break;").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_break_with_argument() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "break 2;").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_continue() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "continue;").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_continue_with_argument() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "continue 2;").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_goto() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "goto label;").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_new() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "new Exception();").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_print() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "print \"Hello\";").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_return() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "return;").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_return_with_argument() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "return 4;").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_throw() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "throw $a;").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_parent() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "parent::a").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_static() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "static::a").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_clone() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "clone $a").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_this() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "$this").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_true() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "true").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_false() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "false").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_null() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "null").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_self() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "self::a").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn test_inline() {
  let arena = bumpalo::Bump::new();
  let asts = parse(&arena, "Hello <?= $world ?>").unwrap();
  insta::assert_yaml_snapshot!(asts);
}
