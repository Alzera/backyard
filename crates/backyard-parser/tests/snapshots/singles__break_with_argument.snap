---
source: crates/backyard-parser/tests/singles.rs
expression: asts
snapshot_kind: text
---
node_type: program
children:
  - node_type: break
    statement:
      node_type: number
      value: "2"
      loc:
        start:
          line: 1
          column: 6
          offset: 6
        end:
          line: 1
          column: 6
          offset: 6
      leadings: ~
      trailings: ~
    loc:
      start:
        line: 1
        column: 0
        offset: 0
      end:
        line: 1
        column: 6
        offset: 6
    leadings: ~
    trailings: ~
loc:
  start:
    line: 1
    column: 0
    offset: 0
  end:
    line: 1
    column: 7
    offset: 7
leadings: ~
trailings: ~
